### ISSUE TITLE
Lab 2 - Set Covering with GA review

### ISSUE THREAD

**Issue by moDal7** [Francesco-Carlucci/ComputationalIntelligence2022#5]
# Lab 2 Review
First of all, thanks for providing such a tidy and efficient solution. I have made my review on the Jupyter notebook version of the code. Your code is very easy to read and understand; I agree that the binary booleans representation of genomes is probably more effective, and I have seen it used in other solutions, but I find that yours is efficient too.

Your choice of hyperparameters seems reasonable; since you got very good results it might be nice to explore around the values, seeing if anything changes with slightly changing the offspring size and the stopping condition.
I particularly liked the way you handled and permitted unfeasible solutions, enabling a possible good but still unfeasible solution to continue its life. 

Only possible suggestions are the Independence between mutation and crossover probability, having then the chance of having them both happening.
Also, a possible self-adaptation system based on the already implemented steadyCnt could be interesting, adjusting the mutation probability or the crossover probability.
Another suggestion could be plotting some results using a graph, to better show the fitness record and maybe the number of fitness calls. 

In conclusion it is a very good solution with extremely good results (w=3304 on N=1000 is probably among the lowest I have seen), so I thank you for posting it!
